sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/v2/Spark33XShims.scala

```diff
@@ -153,6 +157,201 @@ trait Spark33XShims extends Spark33XFileOptionsShims {
     }
     super.tagFileSourceScanExec(meta)
   }
+
+  // 330+ supports YEARMONTH and DAYTIME interval types
+  override def getFileFormats: Map[FileFormatType, Map[FileFormatOp, FileFormatChecks]] = {
+    Map(
+      (ParquetFormatType, FileFormatChecks(
+        cudfRead = (TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.STRUCT +
+            TypeSig.ARRAY + TypeSig.MAP + TypeSig.DAYTIME).nested(),
+        cudfWrite = (TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.STRUCT +
+            TypeSig.ARRAY + TypeSig.MAP + TypeSig.DAYTIME).nested(),
+        sparkSig = (TypeSig.cpuAtomics + TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP +
+            TypeSig.UDT + TypeSig.DAYTIME).nested())))
+  }
+
+  // 330+ supports YEARMONTH and DAYTIME interval types
+  override def getExprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = {
+    val _gpuCommonTypes = TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_64
+    val map: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = Seq(
+      GpuOverrides.expr[Coalesce](
+        "Returns the first non-null argument if exists. Otherwise, null",
+        ExprChecks.projectOnly(
+          (_gpuCommonTypes + TypeSig.DECIMAL_128 + TypeSig.ARRAY + TypeSig.STRUCT +
+              TypeSig.DAYTIME).nested(),
```

Comment:
So using a TypeSig.DAYTIME in versions below 3.3.0 caused issues?  My comment before didn't mean you had to change it. I just want to understand what is happening.

Reply:
```
Strictly speaking:
  DayTimeInterval starts at 320, support DayTimeInterval from/to Parquet starts at 330.
  Coalesce, project and time_add can be put into shims 320Plus.
Putting all the changes into 330 shims is only because I developed and tested these operators(including from/to Parquet) tegather on 330:
  df.coalesce().write("parquet_path")   // involves coalesce
  select timestamp_column + daytime_interval_column  // involves project and time_add
I see main use case of DayTimeInterval in 320 shim is about the literal DayTimeInterval, such as time_add literal day_time_interval. 
Is it necessary adding to 320Plus shims? 
```